deepchem/models/molgan.py

```diff
@@ -0,0 +1,258 @@
+  def iterbatches(epochs):
+      for i in range(epochs):
+          for batch in dataset.iterbatches(batch_size=gan.batch_size, pad_batches=True):
+              adjacency_tensor = one_hot(batch[0], gan.edges)
+              node_tensor = one_hot(batch[1], gan.nodes)
+              yield {gan.data_inputs[0]: adjacency_tensor, gan.data_inputs[1]:node_tensor}
+  gan.fit_gan(iterbatches(8), generator_steps=0.2, checkpoint_interval=5000)
+  generated_data = gan.predict_gan_generator(1000)
+  # convert graphs to RDKitmolecules
+  nmols = feat.defeaturize(generated_data)
+  print("{} molecules generated".format(len(nmols)))
+  # remove invalid moles
+  nmols = list(filter(lambda x: x is not None, nmols))
+  # currently training is unstable so 0 is a common outcome
+  print ("{} valid molecules".format(len(nmols)))
+
+  References
+  ----------
+  .. [1] Nicola De Cao et al. "MolGAN: An implicit generative model
+  for small molecular graphs", https://arxiv.org/abs/1805.11973
+  """
+
+  def __init__(self,
+               edges: int = 5,
+               vertices: int = 9,
+               nodes: int = 5,
+               embedding_dim: int = 10,
+               dropout_rate: float = 0.0,
+               name: str = '',
+               **kwargs):
+    """
+    Initialize the model
+
+    Parameters
+    ----------
+    edges: int, default 5
+        Number of bond types includes BondType.Zero
+    vertices: int, default 9
+        Max number of atoms in adjacency and node features matrices
+    nodes: int, default 5
+        Number of atom types in node features matrix
+    embedding_dim: int, default 10
+        Size of noise input array
+    dropout_rate: float, default = 0.
+        Rate of dropout used across whole model
+    name: str, default ''
+        Name of the model
+    """
+
+    self.edges = edges
+    self.vertices = vertices
+    self.nodes = nodes
+    self.embedding_dim = embedding_dim
+    self.dropout_rate = dropout_rate
+
+    super(BasicMolGANModel, self).__init__(name=name, **kwargs)
+
+  def get_noise_input_shape(self) -> Tuple[int]:
+    """
+    Return shape of the noise input used in generator
+
+    Returns
+    -------
+    Tuple
+        Shape of the noise input
+    """
+
+    return (self.embedding_dim,)
+
+  def get_data_input_shapes(self) -> List:
+    """
+    Return input shape of the discriminator
+
+    Returns
+    -------
+    List
+        List of shapes used as an input for distriminator.
+    """
+    return [(self.vertices, self.vertices, self.edges), (self.vertices,
+                                                         self.nodes)]
+
+  def create_generator(self) -> keras.Model:
+    """
+    Create generator model.
+    Take noise data as an input and processes it through number of
+    dense and dropout layers. Then data is converted into two forms
+    one used for training and other for generation of compounds.
+
+    Returns
+    -------
+    keras.Model
```

Comment:
I'm not sure if this will render correctly in the docs. Could you try building the docs locally and seeing if this looks good?

Reply:
Do you mean running doctest?

Reply:
Hi I tried running sphinx, but struggling to get it working.
Should I just remove Returns part?